buildchain/buildchain/salt_tree.py

```diff
@@ -351,6 +351,7 @@ def task(self) -> types.TaskDict:
         "alertmanager-configuration-secret.sls",
     ),
     Path("salt/metalk8s/addons/prometheus-operator/deployed/chart.sls"),
+    Path("salt/metalk8s/addons/prometheus-operator/deployed/chart-without-grafana.sls"),
```

Comment:
TBH: I'm not sure we need to do this... it means yet another state to maintain (and test),

Since the grafana config can be managed by the user himself, we shouldn't care about it and always deploy grafana. (what happened if we deploy grafana without `auth` section ?)

And maybe it should be the same for the metalk8s-ui depending how the pod behaves when new do not have OIDC (crashLoop or just the web UI that show an error ?)

Reply:
Ok, yes I see, but now to me, we shouldn't deploy Dex as it will not be used, but deploy Grafana and MetalK8s UI even if it's still not usable for the moment... it's ok and easier to maintain

Maybe it should be re-discussed a bit with @scality/metalk8s 

I just do not see a real value of not deploying them. Especially for Grafana where we will need to "maintain" a new salt state (and ensure this one works properly)

Reply:
Already discussed offline on Slack and fixed in this PR + a ticket has been opened for MetalK8s UI